### PR TITLE
JENKINS-47054# Bitbucket api access via http proxy

### PR DESCRIPTION
**Pull request by vivek**
# Description

Adds HTTP proxy support. To simplify code, moved away from fluent http library.

I have tested bitbucket pipeline creation and editing for both bitbucket cloud and server.

See [JENKINS-47054](https://issues.jenkins-ci.org/browse/JENKINS-47054).

# Submitter checklist
- [ ] Link to JIRA ticket in description, if appropriate.
- [ ] Change is code complete and matches issue description
- [ ] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

### CODE REVIEW

File: blueocean-bitbucket-pipeline/src/main/java/io/jenkins/blueocean/blueocean_bitbucket_pipeline/BitbucketApi.java [hunk 1]
```diff
@@ -27,26 +24,19 @@
...
 public abstract class BitbucketApi {
+
     protected final String apiUrl;
-    protected final StandardUsernamePasswordCredentials credentials;
     protected final String userName;
-    protected final String basicAuthHeaderValue;
+    protected final HttpRequest request;
 
     //XXX: To be used for testing to resolve correct factory
     public static final String X_BB_API_TEST_MODE_HEADER="X_BB_API_TEST_MODE_HEADER";
```
Comment (michaelneale) [inline blueocean-bitbucket-pipeline/src/main/java/io/jenkins/blueocean/blueocean_bitbucket_pipeline/BitbucketApi.java]: is this meant to be left in here?

**Comment by michaelneale** [comment]
looks like some testing code still in there - is this to be tested manually only for now?

File: blueocean-bitbucket-pipeline/src/main/java/io/jenkins/blueocean/blueocean_bitbucket_pipeline/BitbucketApi.java (hunk 1 above)
Comment (vivek) [reply blueocean-bitbucket-pipeline/src/main/java/io/jenkins/blueocean/blueocean_bitbucket_pipeline/BitbucketApi.java]: Ah, this is old code put there early on to facilitate wiremock test. I am going to get rid of it.

**Comment by vivek** [comment]
@michaelneale testing is manual. When we have dockerized bitbucket ath it can help avoid it.

**Comment by vivek** [comment]
@michaelneale I have tested it manually for both bitbucket cloud and server. Will be good to have another person test it as well.

**Comment by michaelneale** [comment]
LGTM - @imeredith mind if you have a quick look?

**Pull request merged**